### ISSUE TITLE
Same results on different drcov test

### ISSUE THREAD

**Issue by doria90** [DynamoRIO/dynamorio#2733]
Hi all,
I've been using drcov to record code coverage of Office Word while parsing different files.
I wish to compare both coverage data to see which functions gets executed in the second trial only.
The first file, a DOCX with simple text: `drrun.exe -t drcov -- "C:\Program Files (x86)\Microsoft Office\root\Office16\WINWORD.EXE" "C:\users\home\Desktop\Doc3.docx"`
The second, a DOCX with some functionality: `drrun.exe -t drcov -- "C:\Program Files (x86)\Microsoft Office\root\Office16\WINWORD.EXE" "C:\users\home\Desktop\Doc1.docx"`

I have both files attached here (as Doc1.log and Doc3.log).
In order to compare both executions, I'm using IDA plugin called lighthouse. in lighthouse, I'm getting that both coverages are identical.. 

![image](https://user-images.githubusercontent.com/12772781/33551590-84ba2444-d8fa-11e7-889a-42bacb0bb3d8.png)

Any help?
[Doc1.log](https://github.com/DynamoRIO/dynamorio/files/1526890/Doc1.log)
[Doc3.log](https://github.com/DynamoRIO/dynamorio/files/1526896/Doc3.log)

**Comment by gaasedelen** [comment]
TL;DR: The coverage blocks for `WINWORD.EXE` in the provided `Doc1.log` and `Doc3.log` files are semantically identical. This is _not_ an issue with [Lighthouse](http://github.com/gaasedelen/lighthouse) and I am hesitant to call it an issue with DR/drcov either.

@doria90, do you have anything to indicate that `WINWORD.EXE` is, in fact, executing differently outside of DynamoRIO? If not, I don't think there is an issue with DR/drcov.

Microsoft Office / Word loads 100+ shared libraries (.dll's). I think the issue is that the coverage differences you are looking for are happening in some of those shared libraries. 

Here's a high level output of the two logs:

[Doc1_modules.txt](https://github.com/DynamoRIO/dynamorio/files/1527974/Doc1_modules.txt)
[Doc3_modules.txt](https://github.com/DynamoRIO/dynamorio/files/1527975/Doc3_modules.txt)

**Comment by doria90** [comment]
"Microsoft Office / Word loads 100+ shared libraries (.dll's). I think the issue is that the coverage differences you are looking for are happening in some of those shared libraries."

This line solved it for me. I was (foolishly) trying to use code coverage on WORD.EXE to find a specific function which happens in another DLL.. sorry.
Learned something new though..

Thanks! can close.

**Comment by derekbruening** [comment]
> I'd like to record the coverage with a certain DLL loaded by Word. From what I can tell, it is not possible with drcov.. any suggestions?

Coverage information for all libraries is provided by drcov.  I'm not sure what you're asking?

**Comment by gaasedelen** [comment]
@doria90 the `drcov` command you ran launched Microsoft Word, and collected coverage information for the entire process. This includes the '100+ shared libraries' I mentioned.

The coverage information you want should be in the logs you posted, you just need to open them in a new IDA database created for the library/dll you are interested in.

**Comment by doria90** [comment]
Thanks for replying.
I want to identify the apecific DLL where a certain Word feature is
happening.
Ive prepared Doc1.html witch uses the Word INCLUDETEXT field to read a
local html file and include its content within the document.
Ive also prepared Doc3.html which is a simple document with text.
When recording coverage for both documents, I got the same coverage which
makes no sense as different behavior is obsevered; so I thought I need to
specify to drcov the DLL which is doing the parsing for the coverage.. But
you say drcov is covering all DLLs so Im quite lost..

**Comment by gaasedelen** [comment]
> I want to **identify the apecific DLL** where a certain Word feature is happening.

Unfortunately, Lighthouse doesn't support this. It will help you diff or explore coverage at a module level, but not at a process level. IDA was designed mostly for reversing individual executables, Lighthouse is built ontop of it, with those same principals and expectations.

You will have to pick a module/dll that you think has the functionality/coverage you care about, create an IDA database for it, and then load/diff the coverage files.

> I thought I need to specify to drcov the DLL which is doing the parsing for the coverage.. But you say drcov is covering all DLLs so Im quite lost..

You do not need to specify a target DLL to drcov because it collects coverage for the whole process that gets launched. This likely includes the DLL you care about, one that gets loaded and used by `WINWORD.EXE`

@derekbruening @toshipiazza this issue can probably be closed. I don't see anything actionable from the DR side here.

**Comment by doria90** [comment]
Ok I now understand fully.
Thanks for such a detailed reply!
